main.gs

```diff
@@ -1,40 +1,18 @@
 function doPost(e) { // catches slack slash commands
   if (typeof e !== 'undefined') {
     
-    /// declare variables
-    var globvar = globalVariables();
-    var teamid_true =  globvar['TEAM_ID'];
-    var token_true = PropertiesService.getScriptProperties().getProperty('VERIFICATION_TOKEN'); // expected verification token that accompanies slack API request
-    
     // extract message body
-    var par = e.parameter;
+    var par = e.parameter; 
     
-    // ensure doPost request originates from our slack app - verification token method
-    // note: This is not as secure as using the signed secret method (https://api.slack.com/docs/verifying-requests-from-slack) because the token is not uniquely hashed and can be intercepted. 
-    // note-continued: Think of improving this in the future.
-    var token = par.token;
-  
-    if(!token_true){ // check that token_true has been set in script properties
-      return ContentService.createTextOutput('error: VERIFICATION_TOKEN is not set in script properties. The command will not run. Please contact the web app developer.');
-    }
-    if (token !== token_true) {
-      return ContentService.createTextOutput('error: Invalid token '+token+' . The command will not run. Please contact the web app developer.');
-    }
```

Comment:
Is there a good reason not to keep the token check here? it feels like something we would want to do no matter what input type is.

Reply:
the structure of e.parameter is not the same depending on the type of post request. For slack interactive messages the token is stored in e.parameter.payload.token whereas for slack commands it's stored in e.parameter.token. 

But I agree that having the token check be visible at the very beginning of doPost would be nice. Maybe by making a separate _check_token_ function that handles the various post request structures?

Reply:
Ah - I didn't realise the difference. A separate function sounds like a good idea.